controllers/comments.controller.js

```diff
@@ -1,5 +1,25 @@
-const { selectComments } = require("../models/comments.model");
 const { selectSingleArticle } = require("../models/articles.model");
+const {
+  postCommentOnArticle,
+  checkUserExists,
+} = require("../models/comments.model");
+const { selectComments } = require("../models/comments.model");
+
+exports.postComment = (req, res, next) => {
+  const id = req.params.article_id;
+  const newComment = req.body;
+  const user = req.body.username;
+
+  Promise.all([
+    selectSingleArticle(id),
```

Comment:
These separate check if exists are not necessary as the ID and the user are foreign keys on the comments table, and will result in PSQL errors if they don't exist.